server/server.go

```diff
@@ -79,12 +80,45 @@ var dirPaths = []string{
 	"trie",
 }
 
+// set log output to the designated file
+func setLogFileWriter(logFilePath string) (*os.File, error) {
+	// if file path is empty string return error
+	if logFilePath == "" {
+		return nil, fmt.Errorf("log file path not defined")
+	}
+
+	logFile, err := os.Create(logFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("could not create log file, %w", err)
+	}
+
+	return logFile, nil
+}
+
+// create new logger; log to file if log path is defined
+func newLogger(config *Config) (hclog.Logger, error) {
+	var logFile *os.File
+
+	// if there is an error output logs to console
+	logFile, err := setLogFileWriter(config.LogFilePath)
+	if err != nil {
+		return hclog.New(&hclog.LoggerOptions{
+			Name:  "polygon",
+			Level: config.LogLevel,
+		}), err
+	}
+
+	// write logs to specified file
+	return hclog.New(&hclog.LoggerOptions{
+		Name:   "polygon",
+		Level:  config.LogLevel,
+		Output: logFile,
+	}), nil
+}
```

Comment:
You can rewrite this method to be cleaner like so:
* Have a separate method called `newFileLogger`, which is called from here if the log file path is set (internally it should contain the logic for setLogFileWriter)
* Have a separate method called `newCLILogger`, which just returns
```go
hclog.New(&hclog.LoggerOptions{
			Name:  "polygon",
			Level: config.LogLevel,
		})
```

This way you simplify this method by a lot:
- you're only conditionally calling (and returning) `newFileLogger`
- if that doesn't succeed, just return `newCLILogger`

What do you think?

cc @dbrajovic

Reply:
Agreed. For example:

```
func newLogger(config *Config) (hclog.Logger, error) {
	// log to file if provided
	if fileLogger, err := newFileLogger(config.LogFilePath); err == nil {
		return fileLogger
	}


	// return default cli logger
	return newCLILogger(...)
}
```

Reply:
Checkout new logger logic, much simpler I think fe41b1e3945aa3ff7c35ff6095786ce8d717692c